iothub_client/samples/iothub_client_sample_upload_to_blob_custom/iothub_client_sample_upload_to_blob_custom.c

```diff
@@ -0,0 +1,154 @@
+static char data_to_upload[128];
+
+int main(void)
+{
+    IOTHUB_DEVICE_CLIENT_LL_HANDLE device_ll_handle;
+
+    // Used to initialize IoTHub SDK subsystem
+    (void)IoTHub_Init();
+    (void)printf("Starting the IoTHub client sample upload to blob...\r\n");
+
+    device_ll_handle = IoTHubDeviceClient_LL_CreateFromConnectionString(connectionString, MQTT_Protocol);
+    if (device_ll_handle == NULL)
+    {
+        (void)printf("Failure creating IotHub device. Hint: Check your connection string.\r\n");
+    }
+    else
+    {
+#ifndef WIN32
+        size_t log = 1;
+        (void)IoTHubDeviceClient_LL_SetOption(device_ll_handle, OPTION_CURL_VERBOSE, &log);
+#endif // !WIN32
+
+#ifdef SET_TRUSTED_CERT_IN_SAMPLES
+        // Setting the Trusted Certificate. This is only necessary on systems without
+        // built in certificate stores.
+        IoTHubDeviceClient_LL_SetOption(device_ll_handle, OPTION_TRUSTED_CERT, certificates);
+#endif // SET_TRUSTED_CERT_IN_SAMPLES
+
+        HTTP_PROXY_OPTIONS http_proxy_options = { 0 };
+        http_proxy_options.host_address = proxyHost;
+        http_proxy_options.port = proxyPort;
+
+        if (proxyHost != NULL && IoTHubDeviceClient_LL_SetOption(device_ll_handle, OPTION_HTTP_PROXY, &http_proxy_options) != IOTHUB_CLIENT_OK)
+        {
+            (void)printf("failure to set proxy\n");
+        }
+        else
+        {
+            char* uploadCorrelationId;
+            char* azureBlobSasUri;
+
+            if (IoTHubDeviceClient_LL_InitializeUpload(
+                    device_ll_handle, azureStorageBlobPath, &uploadCorrelationId, &azureBlobSasUri) != IOTHUB_CLIENT_OK)
+            {
+                printf("failed initializing upload in IoT Hub\n");
+            }
+            else
+            {
+                IOTHUB_CLIENT_LL_UPLOADTOBLOB_CONTEXT_HANDLE uploadContextHandle = IoTHubDeviceClient_LL_CreateUploadContext(device_ll_handle, azureBlobSasUri);
+
+                if (uploadContextHandle == NULL)
+                {
+                    (void)printf("failed to create upload context\n");
+                }
+                else
+                {
+                    bool uploadSuccessful = true;
+                    int uploadResultCode = 200;
+
+                    for (uint32_t block_number = 0; block_number < 100; block_number++)
+                    {
+                        int data_size = snprintf(data_to_upload, sizeof(data_to_upload), data_to_upload_format, block_number);
+
+                        if (IoTHubDeviceClient_LL_AzureStoragePutBlock(
+                                uploadContextHandle, block_number, (const uint8_t*)data_to_upload, data_size) != IOTHUB_CLIENT_OK)
+                        {
+                            (void)printf("Failed uploading block number %u to blob. Aborting upload.\n", block_number);
+                            uploadSuccessful = false;
+                            uploadResultCode = 300;
```

Comment:
retry?

Reply:
Yes, this sample will change to have a retry loop here.